oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/GrantedOAuth2AccessToken.java

```diff
@@ -0,0 +1,407 @@
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.EXPIRES_IN;
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.REFRESH_TOKEN;
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.SCOPE;
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.TOKEN_TYPE;
+import static com.linecorp.armeria.internal.common.auth.oauth2.ResponseParserUtil.JSON;
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+import static java.util.Objects.requireNonNull;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.internal.common.auth.oauth2.CaseUtil;
+import com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants;
+
+/**
+ * Defines a structure of the Access Token Response, as per
+ * <a href="https://tools.ietf.org/html/rfc6749#section-5.1">[RFC6749], Section 5.1</a>.
+ */
+@UnstableApi
+public class GrantedOAuth2AccessToken implements Serializable {
+
+    private static final long serialVersionUID = 8698118404098897958L;
+
+    /**
+     * Creates a new {@link GrantedOAuth2AccessToken} based on the {@code JSON}-formatted raw response body and
+     * optional raw formatted {@code scope} used to request the token.
+     * @param rawResponse {@code JSON}-formatted raw response body.
+     * @param requestScope OPTIONAL. A list of space-delimited, case-sensitive strings.
+     *                     The strings are defined by the authorization server.
+     *                     The authorization server MAY fully or partially ignore the scope requested by the
+     *                     client, based on the authorization server policy or the resource owner's
+     *                     instructions. If the issued access token scope is different from the one requested
+     *                     by the client, the authorization server MUST include the "scope" response
+     *                     parameter to inform the client of the actual scope granted.
+     *                     If the client omits the scope parameter when requesting authorization, the
+     *                     authorization server MUST either process the request using a pre-defined default
+     *                     value or fail the request indicating an invalid scope.
+     * @return A new instance of {@link GrantedOAuth2AccessToken}.
+     */
+    public static GrantedOAuth2AccessToken of(String rawResponse, @Nullable String requestScope) {
```

Comment:
Would it be better renaming to `parse(..)`?

Reply:
agree. changed